### PR TITLE
Fix LUT and species clipping

### PR DESCRIPTION
**Pull request by bigfooted**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
 1. arrays in gettriangle need to be size 2 instead of 3
 2. error message that checks species clipping only checks if the keyword is present, not if clipping is on
 3. added setblock_zero "per-species" to CSysVector (added here to reduce the mega-PR for the flamelet model)
 4. fix speling eror

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: Common/include/linear_algebra/CSysVector.hpp [hunk 1]
```diff
@@ -352,6 +352,15 @@ class CSysVector : public VecExpr::CVecExpr<CSysVector<ScalarType>, ScalarType>
     for (auto iVar = 0ul; iVar < nVar; iVar++) vec_val[iPoint * nVar + iVar] = 0.0;
   }
 
+  /*!
+   * \brief Set the value of one variable to zero for one block.
+   * \param[in] iPoint - Index of the block being set to zero.
+   * \param[in] iVar - Index of the variable being set to zero.
+   */
+  inline void SetBlock_Zero(unsigned long iPoint, unsigned long iVar) {
```
Comment (pcarruscag) [inline Common/include/linear_algebra/CSysVector.hpp:360]: This is not needed, just do vector(ipoint, ivar) = 0

File: Common/include/linear_algebra/CSysVector.hpp [hunk 2]
```diff
@@ -352,6 +352,15 @@ class CSysVector : public VecExpr::CVecExpr<CSysVector<ScalarType>, ScalarType>
...
+    vec_val[iPoint * nVar + iVar] = 0.0;
```
Comment (bigfooted) [inline Common/include/linear_algebra/CSysVector.hpp:361]:
```suggestion
    vec_val(iPoint,,iVar) = 0.0;
```

Comment (bigfooted) [reply Common/include/linear_algebra/CSysVector.hpp:361]: Like this, maybe also for the existing SetBlock_Zero?

**Review by pcarruscag — APPROVED** [review]
Yep 👍

**Pull request merged**